### PR TITLE
Clang fixes

### LINKED ISSUE #11187: cmake/toolchain/clang: selecting clang doesn't select the correct compiler file
When setting `ZEPHYR_TOOLCHAIN_VARIANT=clang` and try to compile an example you'll get 
`-- Performing Test toolchain_is_ok - Failed`
This is due to this error:

```
FAILED: cmTC_80be7 
: && /usr/bin/clang --target=i686-pc-none-elf -Dtoolchain_is_ok   CMakeFiles/cmTC_80be7.dir/src.c.obj  -o cmTC_80be7   && :
/usr/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-pc-linux-gnu/8.2.1/../../../libgcc_s.so.1 when searching for libgcc_s.so.1
/usr/bin/ld: skipping incompatible /usr/lib/libgcc_s.so.1 when searching for libgcc_s.so.1
/usr/bin/ld: cannot find libgcc_s.so.1
collect2: error: ld returned 1 exit status
clang-7: error: linker (via gcc) command failed with exit code 1 (use -v to see invocation)
```
This error is caused by the following line: https://github.com/zephyrproject-rtos/zephyr/blob/58781d5782fc7644dafd44d309ef485eee3f88d9/cmake/toolchain.cmake#L71 

due to it trying to compile something called `compiler/clang.cmake` however there is a file called `llvm.cmake` if you change the name of it to `clang.cmake` at least the cmake generation is successful.

Haven't been able to build anything so I wonder a bit about the intention of this `llvm.cmake` file and how clang is supposed to be invoked. Tried for the x86 boards which I assumed is supported from https://github.com/zephyrproject-rtos/zephyr/pull/8520

### PR DESCRIPTION
**Pull request by SebastianBoe**
This PR contains misc. clang fixes and cleanups.

Including;
fixes #11187
resolves https://github.com/zephyrproject-rtos/zephyr/pull/11810#discussion_r239618154

### CODE REVIEW

**Comment by galak** [comment]
@mped-oticon can you take a look?

**Review by galak — APPROVED** [review]

File: cmake/compiler/clang.cmake [hunk 1]
```diff
@@ -9,6 +9,16 @@ if(NOT DEFINED NOSYSDEF_CFLAG)
   set(NOSYSDEF_CFLAG -undef)
 endif()
 
+set(CMAKE_C_COMPILER   ${CLANG_ROOT}/bin/clang)
+set(CMAKE_CXX_COMPILER ${CLANG_ROOT}/bin/clang++)
+set(CMAKE_AR           ${CLANG_ROOT}/bin/llvm-ar		CACHE INTERNAL " " FORCE)
+set(CMAKE_LINKER       ${CLANG_ROOT}/bin/llvm-link		CACHE INTERNAL " " FORCE)
+set(CMAKE_NM           ${CLANG_ROOT}/bin/llvm-nm		CACHE INTERNAL " " FORCE)
+set(CMAKE_OBJDUMP      ${CLANG_ROOT}/bin/llvm-objdump	CACHE INTERNAL " " FORCE)
```
Comment (mped-oticon) [inline cmake/compiler/clang.cmake]: nit: alignment. seems to be tab/space mixed.

File: cmake/compiler/clang.cmake [hunk 2]
```diff
@@ -33,5 +33,20 @@ foreach(isystem_include_dir ${NOSTDINC})
...
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+
+assert_exists(LIBGCC_FILE_NAME)
+
+get_filename_component(LIBGCC_DIR ${LIBGCC_FILE_NAME} DIRECTORY)
+
+assert_exists(LIBGCC_DIR)
+
+list(APPEND LIB_INCLUDE_DIR "-L\"${LIBGCC_DIR}\"")
```
Comment (mped-oticon) [inline cmake/compiler/clang.cmake]: "-L" is not portable. Can't we use link_directories() instead?

**Review by mped-oticon — APPROVED** [review]
Great! A few nits and some portability that can be fixed later.

File: cmake/compiler/clang.cmake (hunk 1 above)
Comment (SebastianBoe) [reply cmake/compiler/clang.cmake]: Fixed in https://github.com/zephyrproject-rtos/zephyr/pull/11810

File: cmake/compiler/clang.cmake (hunk 2 above)
Comment (SebastianBoe) [reply cmake/compiler/clang.cmake]: AFAICT "-L" is portable across clang compilers.

https://clang.llvm.org/docs/ClangCommandLineReference.html#linker-flags

Comment (mped-oticon) [reply cmake/compiler/clang.cmake]: Sorry my mind forgot the context :)

**Pull request merged**